### PR TITLE
chore: refactor datasource storage to use custom repo method

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
As part of transaction support in PG, we are moving from using the jpa methods for database operations. This PR is refactoring the code to use custom repository class for DatasourceStorageRepository from the default CrudRepository.

## Automation

/ok-to-test tags="@tag.ImportExport"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12388971717>
> Commit: 977d50e213b49f5f6180dc8f72326f1fdd6a156a
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12388971717&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.ImportExport`
> Spec:
> <hr>Wed, 18 Dec 2024 09:08:53 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Added methods to retrieve `DatasourceStorage` by `datasourceId` and `environmentId`.
	- Enhanced functionality for fetching `DatasourceStorage` based on `datasourceId`.

- **Bug Fixes**
	- Removed the ability to find a datasource by both its ID and environment ID in the repository interface.

- **Documentation**
	- Updated repository annotations to reflect new functionalities and integrations.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/DatasourceStorageRepositoryCE.java [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
+import com.appsmith.server.repositories.CustomDatasourceStorageRepository;
+import org.springframework.stereotype.Repository;
 import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
 
-public interface DatasourceStorageRepositoryCE extends BaseRepository<DatasourceStorage, String> {
+@Repository
+public interface DatasourceStorageRepositoryCE
+        extends BaseRepository<DatasourceStorage, String>, CustomDatasourceStorageRepository {
     Flux<DatasourceStorage> findByDatasourceId(String datasourceId);
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/DatasourceStorageRepositoryCE.java:12]: Any reason we are not moving above method?

**Review by abhvsn — CHANGES_REQUESTED** [review]
Minor comment.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/DatasourceStorageRepositoryCE.java (hunk 1 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/DatasourceStorageRepositoryCE.java:12]: I missed removing this after the refactor.

Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/DatasourceStorageRepositoryCE.java:12]: Can we do that now?

**Review by abhvsn — APPROVED** [review]

**Pull request merged**